sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormTrainingClient.cs

```diff
@@ -253,19 +243,26 @@ Page<CustomFormModelInfo> NextPageFunc(string nextLink, int? pageSizeHint)
         /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
         /// <returns>A collection of <see cref="CustomFormModelInfo"/> items.</returns>
         [ForwardsClientCalls]
-        public virtual AsyncPageable<CustomFormModelInfo> GetCustomModelsAsync(CancellationToken cancellationToken = default)
+        public virtual Pageable<CustomFormModelInfo> GetCustomModels(CancellationToken cancellationToken = default)
+            => PageableHelpers.CreateEnumerable(GetCustomModelsImplAsync(false, cancellationToken).EnsureSyncEnumerable());
+
+        private async IAsyncEnumerable<Page<CustomFormModelInfo>> GetCustomModelsImplAsync(bool async, [EnumeratorCancellation] CancellationToken cancellationToken)
         {
-            async Task<Page<CustomFormModelInfo>> FirstPageFunc(int? pageSizeHint)
-            {
-                Response<Models_internal> response = await ServiceClient.ListCustomModelsAsync(cancellationToken).ConfigureAwait(false);
-                return Page.FromValues(response.Value.ModelList.Select(info => new CustomFormModelInfo(info)), response.Value.NextLink, response.GetRawResponse());
-            }
-            async Task<Page<CustomFormModelInfo>> NextPageFunc(string nextLink, int? pageSizeHint)
+            Response<Models_internal> response = async
```

Comment:
I personally don't mind having everything merged into one method. We've done that for other FR methods too.
In this case, not sure behavior could change more depending on `async` ...